### PR TITLE
Search for kvm executable too

### PR DESCRIPTION
**Pull request by pilou-**
Not sure if this [fix](https://github.com/lago-project/lago/blob/master/automation/common.sh#L116) is still required.

### CODE REVIEW

**Comment by david-caro** [comment]
ci test please

**Comment by david-caro** [comment]
I think that the fix would still be required, as the issue happens because the libvirt daemon that we use is running out of the chroot where we run lago for the testing, and the binary inside the chroot is not in the same path as outside.

**Comment by david-caro** [comment]
this does not seem to work well on el7:
```
16:15:47 # libvirt: QEMU Driver error : Cannot check QEMU binary /usr/libexec/qemu-kvm: No such file or directory
```

**Comment by pilou-** [comment]
ci test please

**Comment by mykaul** [comment]
@pilou- can you rebase and we'll retest?

**Comment by pilou-** [comment]
@mykaul done :)

**Comment by david-caro** [comment]
ci test please

**Comment by pilou-** [comment]
Instead of hardcoding path of kvm executable, ask libvirt using [capabilities](https://libvirt.org/formatcaps.html).

I am not sure if the case "Nested virtualization not available" should be handled or not: some functional tests could be run without nested virtualization (but not the whole test suite).

Successfully tested on RHEL 7.2 using `patch-only` test suite.

**Comment by pilou-** [comment]
Could you rerun the tests ?

**Comment by eedri** [comment]
ci test please

File: lago/vm.py [hunk 1]
```diff
@@ -226,13 +226,20 @@ def _libvirt_xml(self):
...
-            path
-            for path in [
-                '/usr/libexec/qemu-kvm',
-                '/usr/bin/qemu-kvm',
-            ] if os.path.exists(path)
-        ].pop()
+        capabilities_raw_xml = self.libvirt_con.getCapabilities()
+        capabilities_xml = lxml.etree.fromstring(capabilities_raw_xml)
+        qemu_kvm_path = capabilities_xml.findtext(
+            "guest[os_type='hvm']/arch/domain[@type='kvm']/emulator"
```
Comment (mykaul) [inline lago/vm.py]: Shouldn't the arch be /arch[name='x86_64'] ?

Comment (pilou-) [inline lago/vm.py]: Indeed, `x86_64` is hardcoded in `lago/dom_template.xml`: PR updated using `/arch[@name='x86_64']`.

File: lago/vm.py [hunk 2]
```diff
@@ -226,13 +226,22 @@ def _libvirt_xml(self):
...
-        ].pop()
+        capabilities_raw_xml = self.libvirt_con.getCapabilities()
+        capabilities_xml = lxml.etree.fromstring(capabilities_raw_xml)
+        qemu_kvm_path = capabilities_xml.findtext(
+            "guest[os_type='hvm']/arch[@name='x86_64']/domain[@type='kvm']"
+            "/emulator"
+        )
+
+        if not qemu_kvm_path:
+            # Nested virtualization not available
```
Comment (gbenhaim) [inline lago/vm.py]: This query can tell you if hardware acceleration is enabled, 
I don't see how this is relates to nested virtualization.
Am I missing something ?

Comment (pilou-) [inline lago/vm.py]: If lago is executed inside a VM and if nested virtualization is disabled on the host then this query will return nothing, hence the comment.

But if hardware acceleration is disabled on the host, this query will return nothing too.

I propose to update the comment with: `hardware acceleration (or nested virtualization) not available`, what do you think ?

Comment (gbenhaim) [inline lago/vm.py]: I think that we need to keep only ```hardware acceleration not available```, and also print it to the log.

Comment (pilou-) [inline lago/vm.py]: @gbenhaim ok, PR updated

**Comment by gbenhaim** [comment]
ci test please

File: lago/vm.py (hunk 2 above)
Comment (gbenhaim) [inline lago/vm.py]: +1

**Comment by gbenhaim** [comment]
ci merge please

**Pull request merged**